Controllers/AccountSettingsController.cs

```diff
@@ -10,105 +10,91 @@ public class AccountSettingsController : BaseUserCredentialsController
 
-                userInfo.EmailError = ModelState[_emailErrorKey]?.Errors.FirstOrDefault()?.ErrorMessage;
-                if (userInfo.EmailError != null)
-                    return View(nameof(Index), model);
+            if (userInfo.User.Age == null)
+                userInfo.AgeError = "Age is required";
 
-                await UserService.UpdateUserInfo(userInfo.User);
-                userInfo.SuccessMessage = "Update personal info success!";
+            if (userInfo.EmailError != null || userInfo.AgeError != null)
                 return View(nameof(Index), model);
-            });
+
+            await UserService.UpdateUserInfo(userInfo.User);
+            userInfo.SuccessMessage = "Update personal info success!";
+            return View(nameof(Index), model);
         }
 
         public async Task<IActionResult> SavePassword(AccountSettingsViewModel model)
         {
-            return await RedirectToLogin(async () =>
-            {
-                var changePwd = model.ChangePassword!;
-                model.UserInfo!.User!.Username = HttpContext.Session.GetString(_sessionKeyName)!;
-
-                changePwd.CurrentPasswordError = ModelState[_currentPasswordErrorKey]?.Errors.FirstOrDefault()?.ErrorMessage;
-                changePwd.NewPasswordError = ModelState[_newPasswordErrorKey]?.Errors.FirstOrDefault()?.ErrorMessage;
-                changePwd.RePasswordError = ModelState[_rePasswordErrorKey]?.Errors.FirstOrDefault()?.ErrorMessage;
-                if (changePwd.CurrentPasswordError != null || changePwd.NewPasswordError != null || changePwd.RePasswordError != null)
-                    return View(nameof(Index), model);
+            var changePwd = model.ChangePassword!;
+            model.UserInfo!.User!.Username = HttpContext.User.Identity!.Name!;
 
-                var user = UserService.FindByUsername(model.UserInfo.User.Username);
-                if (user == null)
-                {
-                    return View("Error");
-                }
-                if (user.Password != Encrypter.EncryptSHA256(changePwd.CurrentPassword!))
-                {
-                    changePwd.CurrentPasswordError = "Entered password doesn't match current password";
-                    return View(nameof(Index), model);
-                }
-                if (changePwd.CurrentPassword == changePwd.NewPassword)
-                {
-                    changePwd.NewPasswordError = "New password cannot be the same as current password";
-                    return View(nameof(Index), model);
-                }
-                if (changePwd.NewPassword != changePwd.RePassword)
-                {
-                    changePwd.RePasswordError = changePwd.NewPasswordError = "Passwords do not match";
-                    return View(nameof(Index), model);
-                }
+            changePwd.CurrentPasswordError = ModelState[_currentPasswordErrorKey]?.Errors.FirstOrDefault()?.ErrorMessage;
```

Comment:
Shouldn't define the error for each field, we can use one of 2 options below:
1. Fluent validation
2. Define the Validation attribute for model, then print the whole errors to view using @Html.ValidationSummary(). [See this](https://exceptionnotfound.net/asp-net-mvc-demystified-modelstate/#:~:text=The%20ModelState%20represents%20a%20collection,names%2C%20values%2C%20and%20errors.)